pkg/api/v1/types.go

```diff
@@ -2327,7 +2327,9 @@ type ExportOptions struct {
 // ListOptions is the query options to a standard REST list call.
 type ListOptions struct {
 	unversioned.TypeMeta `json:",inline"`
-
+	// Labels to restrict the list of returned objects by their selector, matched against by pod selectors.
+	// Defaults to match any selector
+	MatchedLabels string `json:"matchedLabels,omitempty"`
```

Comment:
From most concrete out:

I think the type of this should be map[string]string. Unlike for selectors, there's no fear of our labels becoming more complicated than that. So there's not a good reason for presenting this as a string.

Further, I think it should be *map[string]string, to make it clear that it is optional, and to let you explicitly ask for selectors that match things with no labels should you want to do that.

Speculative, but will types ever have multiple selectors (for, e.g., different resources or something)? It seems possible. If that's the case, then we need a map of field name to label set.

Finally, we need a clear name. I think it should have the word "reverse" in it somewhere. ReverseSelectorLookup is my first idea but I think it can be improved.